ui-testing-core/src/main/java/com/avito/android/test/page_object/PageObjectElement.kt

```diff
@@ -12,12 +12,22 @@ import com.avito.android.test.checks.ChecksImpl
 import com.avito.android.test.matcher.NoViewMatcher
 import org.hamcrest.Matcher
 
-open class PageObjectElement(
-    @Deprecated("don't use this matcher directly") val matcher: Matcher<View>,
-    val interactionContext: InteractionContext = SimpleInteractionContext(matcher),
-    open val actions: Actions = ActionsImpl(interactionContext),
-    open val checks: Checks = ChecksImpl(interactionContext)
-) : PageObject(), Actions by actions {
+
+interface PageObjectElement : PageObject, Actions {
+    @Deprecated("don't use this matcher directly")
+    val matcher: Matcher<View>
+    val interactionContext: InteractionContext
+    val actions: Actions
+    val checks: Checks
+}
+
+
+open class BasePageObjectElement(
```

Comment:
Probably, we should name it differently (`ViewPageObjectElement`?)? Because for me it looks a little bit strange when we are instantiating *Base*PageObjectElement :)
Also, what do you think about creating top-level functions like `pageObjectElement(matcher)`?

Reply:
> `ViewPageObjectElement``
Maybe just `ViewElement` for simplicity? We can follow this naming in other cases: `ListElemnt`, `TextViewElement`, `ToolbarElement`, etc

Reply:
> Screen.pageObjectElement() or PageObject.pageObjectElement() extensions for that

But how will we handle case of typed element, not common PageObjectElement? 
For example: 
```
class FooScreen: Screen {
    val title: TextViewElement = ....
}
```